libs/base/Data/These.idr

```diff
@@ -40,6 +40,26 @@ Bifunctor These where
   bimap f g (That b)   = That (g b)
   bimap f g (Both a b) = Both (f a) (g b)
 
+%inline
+public export
+Bifoldable These where
+  bifoldr f _ acc (This a)   = f a acc
+  bifoldr _ g acc (That b)   = g b acc
+  bifoldr f g acc (Both a b) = f a (g b acc)
+
+  bifoldl f _ acc (This a)   = f acc a
+  bifoldl _ g acc (That b)   = g acc b
+  bifoldl f g acc (Both a b) = f (g acc b) a
```

Comment:
Shouldn't this be

```suggestion
  bifoldl f g acc (Both a b) = g (f acc a) b
```

if we're going left-to-right? I expect that depending on whether you implement
`Bifoldable t => t a b -> List (Either a b)` using `bifoldr` or `bifoldl` you
should get one list or its reverse.